native/python/src/fairseq2n/bindings/data/data_pipeline.cc

```diff
@@ -656,7 +656,12 @@ def_data_pipeline(py::module_ &data_module)
     m.def("read_zipped_records", &read_zipped_records, py::arg("path"));
 
     m.def("read_iterator",
-            [](py::iterator iterator, reset_fn fn, bool infinite) {
+            [](py::iterator iterator, reset_fn fn, bool infinite, bool allow_unpickleable) {
+                if (!allow_unpickleable) {
+                    py::gil_scoped_acquire acquire;
+                    py::function pickle_dump_fn = py::module::import("pickle").attr("dumps");
+                    pickle_dump_fn(iterator);
```

Comment:
1. No, the only option seems to be pickling it. (There's a third-party library dill with a `pickles` method to check for pickleability, but even this looks like it performs pickling under the hood). 
Maybe rename `allow_unpickleable` to `skip_pickling_check` or something like this? The user can always set the flag to True if they want to avoid incurring the cost of pickling upon creation.

2. Addressed.

Reply:
Thanks!

1) Yeah, `skip_pickling_check` makes more sense in this case. Can you please add a docstr for this parameter in Python and mention that by default we do pickling during initialization, why we do it, and why one would want to set this to `True`?
2) Thanks!